### ISSUE TITLE
Code in mathematician for check_convergence may not be right

### ISSUE THREAD

**Issue by kwalcock** [Allegra-Cohen/grid#70]
There may be an absolute value or two missing in https://github.com/Allegra-Cohen/grid/blob/8670a1ed8255a770d351efb5885f445d816ddc78/habitus_ui_interface-main/backend/mathematician.py#L287-L294

If a centroid and last_centroid are [0, 0, 0] and [10, 10, 10], then centroid - last_centroid is [-10, -10, -10] with a mean of -10, which is certainly less than 0.00001.

If a centroid and last_centroid are [0, 0, 0] and [10, 0, -10], then centroid - last_centroid is [-10, 0, 10] with a mean of 0, which is also less than 0.00001.

It doesn't seem like that fits the definition of convergence.

The code may have to be

```python
diffs.append(np.abs(np.mean(centroid - last_centroids[i])))
```

for the first issue or even

```python
diffs.append(np.mean(np.abs(centroid - last_centroids[i])))
```
to take care of them both.

**Comment by kwalcock** [comment]
@Allegra-Cohen, in case you are only notified when directly addressed.

**Comment by Allegra-Cohen** [comment]
@kwalcock you're right! I think I threw this function together quite quickly so pretty sure I didn't have a reason to skip abs other than making a mistake. Thanks!
